tools/git-hooks/README.md

```diff
@@ -0,0 +1,91 @@
+<!--
+
+    Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+    
+        http://www.apache.org/licenses/LICENSE-2.0
+    
+    Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+-->
+# Guide to git hooks
+
+The pre-commit hooks enable auto checks before commits. So next time a single `git commit` should do all the style checks automatically.
+
+Currently, the support functionalities are:
+* Turn on/off the hook in local config files.
+* Auto `mvn spotless:apply` before commits
+* Auto `mvn validate` before commits
+* Smart selection of only the changed modules for validation
+
+## Installation
+
+All the files are under `tools/git-hooks`.
+
+### Windows
+
+Double click the `install.ps1`, and allow to execute in admin mode. Make sure that the Git for Windows is the default git on Windows.
+
+**Note**: Windows support is based on the [Git for Windows](https://gitforwindows.org/), which provide a minimal shell environment called Git Bash. Actually, all these hooks on Windows are executed under the Git Bash environment, so the hook script can be somewhat platform-independent.
+
+### Unix
+
+```
+./install.sh
+```
+
+## Configuration
+
+After installation, a `config.sh` shall be created. Modification to the `config.sh` file would take effect the next time you commit.
+
+All the configurable options is in `config.sample.sh`.
+
+```sh
+# enable the pre-commit hooks, 0 - off, 1 - on
+export IOTDB_GIT_HOOKS=1
+# maven path
+# If in Git Bash, replace all '\' to '/', and change the driver path from 'C:\' to '/c/
+# No escape character is needed.
+# Example:
+#   export IOTDB_MAVEN_PATH="/c/Program Files/apache-maven-3.6/bin/mvn"
+export IOTDB_MAVEN_PATH="mvn"
+# git path
+# If in Git Bash, see 'IOTDB_MAVEN_PATH'
+export IOTDB_GIT_PATH="git"
+# smart select the changed java module, 0 - off, 1 - on
+export IOTDB_SMART_JAVA_MODULES=1
+# auto spotless:apply, 0 - off, 1 - on
+export IOTDB_SPOTLESS_APPLY=0
```

Comment:
suggest to enable spotless apply by default (as pre-commit is for simplifying the user behavior, so I think if a developer wants to use this tool, he/she may also want to enable the spotless by default)